Basic-Car-Maintenance/Shared/Odometer/ViewModels/OdometerViewModel.swift

```diff
@@ -0,0 +1,125 @@
+//
+//  OdometerViewModel.swift
+//  Basic-Car-Maintenance
+//
+//  Created by Nate Schaffner on 10/15/23.
+//
+
+import Foundation
+import FirebaseFirestore
+import FirebaseFirestoreSwift
+
+@Observable
+class OdometerViewModel {
+    
+    let authenticationViewModel: AuthenticationViewModel
+    
+    var readings = [OdometerReading]()
+    var showAddErrorAlert = false
+    var showErrorAlert = false
+    var isShowingAddOdometerReading = false
+    var isShowingAddView = false
+    var errorMessage: String = ""
+    var sortOption: SortOption = .custom
+    var vehicles = [Vehicle]()
+
+    init(authenticationViewModel: AuthenticationViewModel) {
+        self.authenticationViewModel = authenticationViewModel
+    }
+    
+    func addReading(_ odometerReading: OdometerReading) {
+        if let uid = authenticationViewModel.user?.uid {
+            var readingToAdd = odometerReading
+            readingToAdd.userID = uid
+            
+            do {
+                let documentReference = try Firestore
+                    .firestore()
+                    .collection("odometer_readings")
+                    .addDocument(from: readingToAdd)
+                
+                var reading = odometerReading
+                let documentId = documentReference.documentID
+                reading.id = documentId
+                
+                readings.append(reading)
+                
+                errorMessage = ""
+                isShowingAddOdometerReading = false
+            } catch {
+                showAddErrorAlert.toggle()
+                errorMessage = error.localizedDescription
+            }
+        }
+    }
+    
+    func getOdometerReadings() async {
+        if let uid = authenticationViewModel.user?.uid {
+            let db = Firestore.firestore()
+            let docRef = db.collection("odometer_readings").whereField("userID", isEqualTo: uid)
```

Comment:
Same with the enum `FirestoreCollection ` again here, and the `userID`

Reply:
fixed in https://github.com/mikaelacaron/Basic-Car-Maintenance/pull/175/commits/ae3b547c60f52d1546312b55f241de4770f78149